google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/main/java/com/google/cloud/spanner/jdbc/StatementResultImpl.java

```diff
@@ -0,0 +1,187 @@
+
+  /**
+   * {@link StatementResult} containing a {@link ResultSet} created by a {@link
+   * ClientSideStatement}.
+   */
+  static StatementResult of(ResultSet resultSet, ClientSideStatementType clientSideStatementType) {
+    return new StatementResultImpl(resultSet, clientSideStatementType);
+  }
+
+  /** {@link StatementResult} containing an update count returned by Cloud Spanner. */
+  static StatementResult of(Long updateCount) {
+    return new StatementResultImpl(updateCount);
+  }
+
+  /**
+   * Convenience method for creating a {@link StatementResult} containing a {@link ResultSet} with
+   * one BOOL column and one row that is created by a {@link ClientSideStatement}.
+   */
+  static StatementResult resultSet(
+      String name, Boolean value, ClientSideStatementType clientSideStatementType) {
+    return of(
+        ResultSets.forRows(
+            Type.struct(StructField.of(name, Type.bool())),
+            Arrays.asList(Struct.newBuilder().set(name).to(value).build())),
+        clientSideStatementType);
+  }
+
+  /**
+   * Convenience method for creating a {@link StatementResult} containing a {@link ResultSet} with
+   * one INT64 column and one row that is created by a {@link ClientSideStatement}.
+   */
+  static StatementResult resultSet(
+      String name, Long value, ClientSideStatementType clientSideStatementType) {
+    return of(
+        ResultSets.forRows(
+            Type.struct(StructField.of(name, Type.int64())),
+            Arrays.asList(Struct.newBuilder().set(name).to(value).build())),
+        clientSideStatementType);
+  }
+
+  /**
+   * Convenience method for creating a {@link StatementResult} containing a {@link ResultSet} with
+   * one ARRAY<INT64> column and one row that is created by a {@link ClientSideStatement}.
+   */
+  static StatementResult resultSet(
+      String name, long[] values, ClientSideStatementType clientSideStatementType) {
+    return of(
+        ResultSets.forRows(
+            Type.struct(StructField.of(name, Type.array(Type.int64()))),
+            Arrays.asList(Struct.newBuilder().set(name).toInt64Array(values).build())),
+        clientSideStatementType);
+  }
+
+  /**
+   * Convenience method for creating a {@link StatementResult} containing a {@link ResultSet} with
+   * one STRING column and one row that is created by a {@link ClientSideStatement}.
+   */
+  static StatementResult resultSet(
+      String name, String value, ClientSideStatementType clientSideStatementType) {
+    return of(
+        ResultSets.forRows(
+            Type.struct(StructField.of(name, Type.string())),
+            Arrays.asList(Struct.newBuilder().set(name).to(value).build())),
+        clientSideStatementType);
+  }
+
+  /**
+   * Convenience method for creating a {@link StatementResult} containing a {@link ResultSet} with
+   * one STRING column containing an {@link Enum} value and one row that is created by a {@link
+   * ClientSideStatement}.
+   */
+  static StatementResult resultSet(
```

Comment:
do we need more tests here for all the different types?

Reply:
I've added test cases for all the missing convenience methods.